test/ui_test_suite.rb

```diff
@@ -3504,6 +3515,234 @@ def open_new_page(url)
 		assert (plot_labels.include? "user-#{$random_seed}: group0 (3 points)"), "labels are incorrect: '#{plot_labels}' should include 'user-#{$random_seed}: group0'"
 	end
 
+	# make sure sharing the annotation works
+	test 'front-end: user annotation sharing' do
+		puts "Test method: #{self.method_name}"
+
+		# login
+		login_path = @base_url + '/users/sign_in'
+		@driver.get login_path
+		wait_until_page_loads(login_path)
+		login($test_email)
+
+		# load annotation panel
+		annot_path = @base_url + '/user_annotations'
+		@driver.get annot_path
+
+		@driver.find_element(:class, "user-#{$random_seed}-exp-edit").click
+		wait_until_page_loads('edit user annotation path')
+
+		# click the share button
+		share_button = @driver.find_element(:id, 'add-user-annotation-share')
+		share_button.click
+
+		share_email = @driver.find_element(:class, 'share-email')
+		share_email.send_keys($share_email)
+
+		share_permission = @driver.find_element(:class, 'share-permission')
+		share_permission.send_keys('Edit')
+
+		#update the annotation
+		submit = @driver.find_element(:id, 'submit-button')
+		submit.click
+
+		wait_until_page_loads('user annotation path')
+
+		# logout
+		wait_for_render(:id, 'message_modal')
+		close_modal('message_modal')
+		profile = @driver.find_element(:id, 'profile-nav')
+		profile.click
+		logout = @driver.find_element(:id, 'logout-nav')
+		logout.click
+		wait_until_page_loads(@base_url)
+		close_modal('message_modal')
+
+		# login
+		login_as_other($test_email)
+		# load annotation panel
+		annot_path = @base_url + '/user_annotations'
+		@driver.get annot_path
+
+		#View the annotation
+		wait_until_page_loads('view user annotation index')
+		@driver.find_element(:class, "user-#{$random_seed}-exp-show").click
+		wait_until_page_loads('view user annotation path')
+
+		#make sure the new annotation still renders a plot for plotly
+		@wait.until {wait_for_plotly_render('#cluster-plot', 'rendered')}
+		annot_rendered = @driver.execute_script("return $('#cluster-plot').data('rendered')")
+		assert annot_rendered, "cluster plot did not finish rendering on annotation change, expected true but found #{annot_rendered}"
+
+		# load random gene to search
+		gene = @genes.sample
+		search_box = @driver.find_element(:id, 'search_genes')
+		search_box.send_key(gene)
+		wait_for_render(:id, 'perform-gene-search')
+		search_genes = @driver.find_element(:id, 'perform-gene-search')
+		search_genes.click
+
+		#make sure the new annotation still renders plots for plotly
+		assert element_present?(:id, 'box-controls'), 'could not find expression violin plot'
+		assert element_present?(:id, 'scatter-plots'), 'could not find expression scatter plots'
+
```

Comment:
rather than calling sleep, use wait_for_render or some other boolean declaration to hold execution.